packages/@lwc/engine/src/framework/def.ts

```diff
@@ -127,43 +113,23 @@ function createComponentDef(
     const superDef: ComponentDef | null =
         (superProto as any) !== BaseLightningElement
             ? getComponentDef(superProto, subclassComponentName)
-            : null;
+            : lightingElementDef;
     const SuperBridge = isNull(superDef) ? BaseBridgeElement : superDef.bridge;
-    const bridge = HTMLBridgeElementFactory(
-        SuperBridge,
-        getOwnPropertyNames(props),
-        getOwnPropertyNames(methods)
-    );
-    if (!isNull(superDef)) {
-        props = assign(create(null), superDef.props, props);
-        methods = assign(create(null), superDef.methods, methods);
-        wire = superDef.wire || wire ? assign(create(null), superDef.wire, wire) : undefined;
-        track = assign(create(null), superDef.track, track);
-        connectedCallback = connectedCallback || superDef.connectedCallback;
-        disconnectedCallback = disconnectedCallback || superDef.disconnectedCallback;
-        renderedCallback = renderedCallback || superDef.renderedCallback;
-        errorCallback = errorCallback || superDef.errorCallback;
-        render = render || superDef.render;
-        template = template || superDef.template;
-    }
-    props = assign(create(null), HTML_PROPS, props);
-
-    if (!isUndefined(fields)) {
```

Comment:
@caridy this is the only place we can mark each field as observable, because at compile time (or when the `registerDecorators` is called) we dont know if the class is a lwc component.